pkg/controller/deployment/deployment_controller.go

```diff
@@ -718,14 +718,22 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment, maxOl
 	}
 	// Set new RC's annotation
 	setNewRCAnnotations(&deployment, &newRC, newRevision)
+	allRCs := append(oldRCs, &newRC)
+	newReplicasCount, err := deploymentutil.NewRCNewReplicas(&deployment, allRCs, &newRC)
+	if err != nil {
+		return nil, err
+	}
+	newRC.Spec.Replicas = newReplicasCount
 	createdRC, err := dc.client.Core().ReplicationControllers(namespace).Create(&newRC)
 	if err != nil {
 		dc.rcExpectations.DeleteExpectations(dKey)
 		return nil, fmt.Errorf("error creating replication controller: %v", err)
 	}
+	if newReplicasCount > 0 {
+		dc.eventRecorder.Eventf(&deployment, api.EventTypeNormal, "ScalingRC", "Scaled %s rc %s to %d", "up", createdRC.Name, newReplicasCount)
+	}
 
-	err = dc.updateDeploymentRevision(deployment, newRevision)
-	return createdRC, err
+	return createdRC, dc.updateDeploymentRevision(deployment, newRevision)
```

Comment:
Deployment revision is synced periodically right after getting new RS. 
https://github.com/janetkuo/kubernetes/blob/deployment-new-rc-replicas/pkg/controller/deployment/deployment_controller.go#L599-L605
```go
	// Sync deployment's revision number with new replica set
	if newRS.Annotations != nil && len(newRS.Annotations[deploymentutil.RevisionAnnotation]) > 0 &&
		(deployment.Annotations == nil || deployment.Annotations[deploymentutil.RevisionAnnotation] != newRS.Annotations[deploymentutil.RevisionAnnotation]) {
		if err = dc.updateDeploymentRevision(deployment, newRS.Annotations[deploymentutil.RevisionAnnotation]); err != nil {
			glog.V(4).Infof("Error: %v. Unable to update deployment revision, will retry later.", err)
		}
	}
```